packages/style-engine/class-wp-style-engine.php

```diff
@@ -507,19 +625,67 @@ protected static function get_individual_property_css_declarations( $style_value
+ *     'convert_vars_to_classnames' => (boolean) Whether to skip converting CSS var:? values to var( --wp--preset--* ) values. Default is `false`.
+ *     'enqueue'                    => (boolean) When `true`, will enqueue the parsed styles for rendering on the frontend in a block-supports inline style tag. Default is `false`.
+ * );.
  *
- * @return array<string>|null array(
- *     'css'          => (string) A CSS ruleset or declarations block formatted to be placed in an HTML `style` attribute or tag.
- *     'declarations' => (array) An array of property/value pairs representing parsed CSS declarations.
- *     'classnames'   => (string) Classnames separated by a space.
+ * @return array<string> array(
+ *     'css'           => (string) A CSS ruleset or declarations block formatted to be placed in an HTML `style` attribute or tag.
+ *     'declarations'  => (array) An array of property/value pairs representing parsed CSS declarations.
+ *     'classnames'    => (string) Classnames separated by a space.
  * );
  */
 function wp_style_engine_get_block_supports_styles( $block_styles, $options = array() ) {
 	if ( class_exists( 'WP_Style_Engine' ) ) {
-		$style_engine = WP_Style_Engine::get_instance();
-		return $style_engine->get_block_supports_styles( $block_styles, $options );
+		$defaults = array(
+			'selector'                   => null,
+			'convert_vars_to_classnames' => false,
+			'enqueue'                    => false,
+		);
+
+		$options       = wp_parse_args( $options, $defaults );
+		$style_engine  = WP_Style_Engine::get_instance();
+		$parsed_styles = $style_engine->parse_block_supports_styles( $block_styles, $options );
+
+		// Output.
+		$styles_output = array();
+		if ( ! empty( $parsed_styles['css_declarations'] ) ) {
+			$styles_output['css']          = $style_engine->compile_css( $parsed_styles['css_declarations'], $options['selector'] );
+			$styles_output['declarations'] = $parsed_styles['css_declarations'];
+			if ( true === $options['enqueue'] ) {
+				$style_engine->store_css_rule( $options['selector'], $parsed_styles['css_declarations'], 'block-supports' );
+			}
+		}
+
+		if ( ! empty( $parsed_styles['classnames'] ) ) {
+			$styles_output['classnames'] = $style_engine->compile_classnames( $parsed_styles['classnames'] );
+		}
+
+		return array_filter( $styles_output );
+	}
+	return array();
+}
+
+/**
+ * Global public interface method to parse block styles from a single block style object
+ * and then, via the `enqueue` flag, will enqueue them for rendering on the frontend in a block-supports inline style tag.
+ *
+ * @access public
+ *
+ * @param string $selector         A CSS selector.
+ * @param array  $css_declarations An array of CSS definitions, e.g., array( "$property" => "$value" ).
+ *
+ * @return boolean Whether the storage process was successful.
+ */
+function wp_style_engine_enqueue_block_supports_styles( $selector, $css_declarations ) {
```

Comment:
So this is the method I was thinking layout could use, given that the layout styles don't need any block styles parsing.